File: KBDTypeServer/Application/DTOs/AuthDtos/RegistrationDto.cs
using System.ComponentModel.DataAnnotations;

namespace KBDTypeServer.Application.DTOs.AuthDtos
{
    public class RegistrationDto
    {
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [Required]
        [RegularExpression(@"^380\d{9}$", ErrorMessage = "Phone number must start with 380 and contain 9 digits after.")]
        public string PhoneNumber { get; set; }
        public string? Email { get; set; }
        [Required]
        public string Password { get; set; }
    }
}

File: KBDTypeFront/src/dto/auth/RegistrationDto.ts
// dto/RegistrationDto.ts
export class RegistrationDto {
  firstName: string = "";      // Ім'я (обов'язково, не менше 1 символу)
  lastName: string = "";       // Прізвище (обов'язково, не менше 1 символу)
  phoneNumber: string = "";    // Телефон (обов'язково, формат ^380\d{9}$)
  email?: string | null = null;  // Email (необов'язково)
  password: string = "";       // Пароль (обов'язково, не менше 1 символу)

  constructor(init?: Partial<RegistrationDto>) {
    Object.assign(this, init);
  }
}